packages/squiggle-lang/src/rescript/Reducer/Reducer_Dispatch/Reducer_Dispatch_BuiltIn.res

```diff
@@ -1,72 +1,72 @@
-module CTV = Reducer_Extension.CodeTreeValue
-module Lib = Reducer_Extension.ReducerLibrary
-module ME = Reducer_MathJs.Eval
-module Rerr = Reducer_Error
+module ExternalLibrary = ReducerInterface.ExternalLibrary
+module MathJs = Reducer_MathJs
+open ReducerInterface.ExpressionValue
+open Reducer_ErrorValue
+
 /*
   MathJs provides default implementations for builtins
   This is where all the expected builtins like + = * / sin cos log ln etc are handled
   DO NOT try to add external function mapping here!
 */
-type codeTreeValue = CTV.codeTreeValue
-type reducerError = Rerr.reducerError
 
 exception TestRescriptException
 
-let callInternal = (call: CTV.functionCall): result<'b, reducerError> =>{
-
-  let callMatjJs = (call: CTV.functionCall): result<'b, reducerError> =>
+let callInternal = (call: functionCall): result<'b, errorValue> => {
+  let callMathJs = (call: functionCall): result<'b, errorValue> =>
     switch call {
-      | ("jsraise", [msg]) => Js.Exn.raiseError(CTV.show(msg)) // For Tests
-      | ("resraise", _) => raise(TestRescriptException) // For Tests
-      | call => call->CTV.showFunctionCall-> ME.eval
+    | ("jsraise", [msg]) => Js.Exn.raiseError(show(msg)) // For Tests
+    | ("resraise", _) => raise(TestRescriptException) // For Tests
+    | call => call->showFunctionCall->MathJs.Eval.eval
     }
 
   let constructRecord = arrayOfPairs => {
-      Belt.Array.map(arrayOfPairs, pairValue => {
+    Belt.Array.map(arrayOfPairs, pairValue => {
       switch pairValue {
-      | CTV.CtvArray([CTV.CtvString(key), valueValue]) =>
-        (key, valueValue)
-      | _ => ("wrong key type", pairValue->CTV.showWithType->CTV.CtvString)}
-      }) -> Js.Dict.fromArray -> CTV.CtvRecord -> Ok
+      | EvArray([EvString(key), valueValue]) => (key, valueValue)
+      | _ => ("wrong key type", pairValue->showWithType->EvString)
+      }
+    })
+    ->Js.Dict.fromArray
+    ->EvRecord
+    ->Ok
   }
 
-  let arrayAtIndex = (aValueArray: array<codeTreeValue>, fIndex: float) =>
+  let arrayAtIndex = (aValueArray: array<expressionValue>, fIndex: float) =>
     switch Belt.Array.get(aValueArray, Belt.Int.fromFloat(fIndex)) {
-    | Some(value) => value -> Ok
-    | None => Rerr.RerrArrayIndexNotFound("Array index not found", Belt.Int.fromFloat(fIndex)) -> Error
+    | Some(value) => value->Ok
+    | None => REArrayIndexNotFound("Array index not found", Belt.Int.fromFloat(fIndex))->Error
```

Comment:
Minor note, but why leave the string as a parameter, when this message would always be the case for REArrayIndexNotFound? You could just have it in the REArrayIndexNotFound toString method

Reply:
Those are redundant default messages. If you don't feel like writing your own message yet.

Reply:
RESOLVED